### ISSUE TITLE
generated project's repo/pom.xml contains dependency to repo-amp (share, too)

### ISSUE THREAD

**Issue by michaeljmuller** [binduwavell/generator-alfresco#10]
The generated project moved repo-amp to amps_source_templates, but left the reference to those projects in the repo/pom.xml and share/pom.xml.

**Comment by binduwavell** [comment]
Which options did you select in the generator? You can just attach your .yo-rc.json to answer this question if you like :)

**Comment by michaeljmuller** [comment]
{
  "generator-alfresco": {
    "originalGeneratorVersion": "0.0.1",
    "generatorVersion": "0.0.1",
    "sdkVersion": "2.1.1",
    "projectGroupId": "com.sj",
    "projectArtifactId": "ecm",
    "projectVersion": "1.0.0-SNAPSHOT",
    "projectPackage": "com.sj",
    "communityOrEnterprise": "Enterprise",
    "includeGitIgnore": true,
    "removeDefaultSourceAmps": true,
    "removeDefaultSourceSamples": false,
    "archetypeVersion": "2.1.1",
    "moduleRegistry": []
  }
}

**Comment by binduwavell** [comment]
After discussing in person, the removeDefaultSourceAmps: true setting caused the amps to be removed. You then attempted to build the project without adding any additional amps and it failed because we did not remove the references somewhere. I just reproduced this and got the following error:

```
[ERROR] Failed to execute goal on project repo: Could not resolve dependencies for project com.sj:repo:war:1.0.0-SNAPSHOT: Could not find artifact com.sj:repo-amp:amp:1.0.0-SNAPSHOT in alfresco-public (https://artifacts.alfresco.com/nexus/content/groups/public) -> [Help 1]
```

I'm not sure why, but the default repo-amp is not removed from repo/pom.xml and the default share-amp is not removed from the share/pom.xml. We need to add tests for this and then fix the problem.
